src/golang/lib/workflow/utils/database.go

```diff
@@ -374,6 +376,57 @@ func UpdateDAGResultMetadata(
 		return err
 	}
 
+	// Check if DAG has finished running.
```

Comment:
can you update the comment for this function so it describes the fact that the artifact and op results are also updated if the dag status changes to cancelled or failed